com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/JdtSourceLookUpProvider.java

```diff
@@ -155,23 +156,42 @@ public String[] getFullyQualifiedName(String uri, int[] lines, int[] columns) th
         String[] fqns = new String[lines.length];
         if (astUnit != null) {
             for (int i = 0; i < lines.length; i++) {
-                // TODO
-                // The ValidBreakpointLocationLocator will verify if the current line is a valid location or not.
-                // If so, it will return the fully qualified name of the class type that contains the current line.
-                // Otherwise, it will try to find a valid location from the next lines and return it's fully qualified name.
-                // In current stage, we don't support to move the invalid breakpoint down to the next valid location, and just
-                // mark it as "unverified".
-                // In future, we could consider supporting to update the breakpoint to a valid location.
-                BreakpointLocationLocator locator = new BreakpointLocationLocator(astUnit, lines[i], true, true);
-                astUnit.accept(locator);
-                // When the final valid line location is same as the original line, that represents it's a valid breakpoint.
-                // Add location type check to avoid breakpoint on method/field which will never be hit in current implementation.
-                if (lines[i] == locator.getLineLocation()
-                        && locator.getLocationType() == BreakpointLocationLocator.LOCATION_LINE) {
-                    fqns[i] = locator.getFullyQualifiedTypeName();
-                } else if (locator.getLocationType() == BreakpointLocationLocator.LOCATION_METHOD) {
-                    fqns[i] = locator.getFullyQualifiedTypeName().concat("#").concat(locator.getMethodName())
-                            .concat("#").concat(locator.getMethodSignature());
+                // if we have a column, try to find the lambda expression at that column
+                LambdaExpressionLocator lambdaExpressionLocator = new LambdaExpressionLocator(astUnit, lines[i],
```

Comment:
If it's not an inline breakpoint (e.g. column number is -1), we can just skip the LambdaVisitor here.

Reply:
True my bad, I was planning to skip the unwanted visiting. I will fix it.